src/dcos_e2e/backends/_aws/__init__.py

```diff
@@ -0,0 +1,333 @@
+        )
+
+        # Also DcosCloudformationLauncher
+        # Get a OnpremLauncher object
+        self.launcher = get_launcher(  # type: ignore
+            config=validated_launch_config,
+        )
+
+        # Create the AWS stack from the DcosCloudformationLauncher.
+        # Update ``cluster_info`` with the AWS SSH key information.
+        self.cluster_info = self.launcher.create()
+
+        # Store the generated AWS SSH key to the file system.
+        self._ssh_key_path = self._path / 'id_rsa'
+        private_key = self.cluster_info['ssh_private_key']
+        Path(self._ssh_key_path).write_bytes(private_key.encode())
+
+        # Wait for the AWS stack setup completion.
+        DcosCloudformationLauncher.wait(self.launcher)  # type: ignore
+
+        # Update the cluster_info with AWS stack information:
+        # ``describe`` fetches the latest information for the stack.
+        # This makes node IP addresses available to ``cluster_info``.
+        # This also inserts bootstrap node information into ``cluster_info``.
+        self.cluster_info = AbstractOnpremLauncher.describe(  # type: ignore
+            self.launcher,
+        )
+
+    def install_dcos_from_url(
+        self,
+        build_artifact: str,
+        extra_config: Dict[str, Any],
+        log_output_live: bool,
+    ) -> None:
+        """
+        Install DC/OS from a URL.
+
+        Args:
+            build_artifact: The URL string to a build artifact to install DC/OS
+                from.
+            extra_config: This may contain extra installation configuration
+                variables that are applied on top of the default DC/OS
+                configuration of the AWS backend.
+            log_output_live: If ``True``, log output of the installation live.
+
+        Raises:
+            KeyboardInterrupt: If a keyboard interrupt is triggered during the
+                installation of DC/OS.
+            Exception: If the DC/OS installation fails for any other reason.
+
+        """
+        # In order to install DC/OS with the preliminary dcos-launch
+        # config the ``build_artifact`` URL is overwritten.
+        self.launcher.config['installer_url'] = build_artifact
+
+        # The DC/OS config parameters from ``extra_config`` are applied
+        # on top of the preliminary DC/OS config.
+        dcos_config = self.launcher.config['dcos_config']
+        self.launcher.config['dcos_config'] = {**dcos_config, **extra_config}
+
+        # The ``wait`` method starts the actual DC/OS installation process.
+        # We do not use ``self.launcher.wait()`` here because at the time of
+        # writing it does both, waiting for the AWS stack and installing
+        # DC/OS. This is desired to be changed by the dcos-launch team.
+        # https://jira.mesosphere.com/browse/DCOS-21660
+        try:
+            AbstractOnpremLauncher.wait(self.launcher)  # type: ignore
+        except (KeyboardInterrupt, Exception):
```

Comment:
Is it possible that someone will want to interrupt the install of DC/OS and not destroy the cluster?

What do you think of moving this to `cluster.py`? In a separate PR?

Reply:
This could indeed be moved to cluster.py.
A half-provisioned cluster remains in an undefined state and from my point of view is useless to keep around.